src/components/atoms/StyledInput.tsx

```diff
@@ -0,0 +1,27 @@
+import styled, { css } from 'styled-components';
+
+type InputProps = {
+  submit?: boolean;
+  theme: any;
+};
+
+export const Input = styled.input<InputProps>`
+  box-sizing: border-box;
+  padding: 0.75em;
+  margin: 0.5em;
+  border: 1px solid ${props => props.theme.primary};
+  color: ${props => props.theme.darkAccent};
+  background: ${props => props.theme.lightText};
+  border-radius: 5px;
+
+  ${(props: InputProps) =>
+    props.submit &&
+    css`
+      background: ${inputProps => inputProps.theme.primary};
```

Comment:
These properties are already generated within a function and `props` is in scope, so the extra function with `inputProps` should be unnecessary here.